services/shhext/api.go

```diff
@@ -461,33 +488,23 @@ func makeEnvelop(
 }
 
 // makePayload makes a specific payload for MailServer to request historic messages.
-func makePayload(r MessagesRequest) []byte {
-	// Payload format:
-	// 4  bytes for lower
-	// 4  bytes for upper
-	// 64 bytes for the bloom filter
-	// 4  bytes for limit
-	// 36 bytes for the cursor. optional.
-	data := make([]byte, 12+whisper.BloomFilterSize)
-
-	// from
-	binary.BigEndian.PutUint32(data, r.From)
-	// to
-	binary.BigEndian.PutUint32(data[4:], r.To)
-	// bloom
-	copy(data[8:], createBloomFilter(r))
-	// limit
-	binary.BigEndian.PutUint32(data[8+whisper.BloomFilterSize:], r.Limit)
-
-	// cursor is the key of an envelope in leveldb.
-	// it's 36 bytes. 4 bytes for the timestamp + 32 bytes for the envelope hash
+func makePayload(r MessagesRequest) ([]byte, error) {
 	expectedCursorSize := common.HashLength + 4
-	cursorBytes, err := hex.DecodeString(r.Cursor)
-	if err != nil || len(cursorBytes) != expectedCursorSize {
-		return data
+	cursor, err := hex.DecodeString(r.Cursor)
+	if err != nil || len(cursor) != expectedCursorSize {
+		cursor = nil
+	}
+
+	payload := MessagesRequestPayload{
+		Lower:  r.From,
+		Upper:  r.To,
+		Bloom:  createBloomFilter(r),
+		Limit:  r.Limit,
+		Cursor: cursor,
+		Batch:  r.Batch,
```

Comment:
are you sure that it is necessary for a client to know that envelopes are fetched in batches? this is more of a transport layer detail

Reply:
Unfortunately yes. It's the client code that supports or does not support receiving envelopes in batches. The other option was to introduce a method as you mentioned in the other PR.

We will be able to get rid of this property after two releases.

Also, I guess we can hide it from the client... We can set it to `true` by default in this PR.

Reply:
Yes, I changed it exactly this way.